set.mm

```diff
@@ -735799,6 +735836,340 @@ their group (addition) operations are equal for all pairs of elements of
+      ( vi cn wcel wa cfv c1 cseq co adantr cv cplusg csn cxp cuz elnnuz biimpi
+      cgsu cfz cmpt wceq a1i eqidd simpr fvmptd elfznn fvconst2g syl2an seqfveq
+      weq eqtr4d cvv eqid elfvex eleq2s adantl fmptd gsumval2 mulgnn 3eqtr4rd
+      cbs ) FLMZGBMZNZFEUAOZDPQOFVNLGUBUCZPQZOEDUGRFGCRVMVNKDVOPFVKFPUDOMZVLVKV
+      QFUEUFSZVMKTZPFUHRZMZNZVSDOGVSVOOZWBAVSGGVTDBDAVTGUIUJWBJUKWBAKUSNGULVMWA
+      UMVMVLWAVKVLUMZSUNVMVLVSLMWCGUJWAWDVSFUOLGVSBUPUQUTURVMBVNDEPFVAHVNVBZVLE
+      VAMZVKWFGEVJOBGEVJVCHVDVEVRVMAVTGBDVMVLATVTMWDSJVFVGBVNVPCEFGHWEIVPVBVHVI
+      $.
+
+    $( Group multiple (exponentiation) operation at a nonnegative integer
+       expressed by a groups sum.  This corresponds to the definition in [Lang]
+       p. 6, second formula.  (Contributed by AV, 28-Dec-2023.) $)
+    mulgnn0gsum $p |- ( ( N e. NN0 /\ X e. B )
+                        -> ( N .x. X ) = ( G gsum F ) ) $=
+      ( wcel co cgsu wceq cc0 ex c0 c1 cfz syl6eq cn0 cn wo wi elnn0 mulgnngsum
+      wa c0g cfv oveq1 eqid mulg0 sylan9eq cmpt oveq2 fz10 eqidd mpteq12dv mpt0
+      syl5eq adantr oveq2d gsum0 eqtr4d jaoi sylbi imp ) FUAKZGBKZFGCLZEDMLZNZV
+      HFUBKZFONZUCVIVLUDZFUEVMVOVNVMVIVLABCDEFGHIJUFPVNVIVLVNVIUGZVJEUHUIZVKVNV
+      IVJOGCLVQFOGCUJBCEGVQHVQUKZIULUMVPVKEQMLVQVPDQEMVNDQNVIVNDARFSLZGUNZQJVNV
+      TAQGUNQVNAVSGQGVNVSROSLQFORSUOUPTVNGUQURAGUSTUTVAVBEVQVRVCTVDPVEVFVG $.
+  $}
+
+  ${
+    $d A x $.  $d F i $.  $d X i $.  $d i x $.  $d .x. x $.
+    cycsubm.b $e |- B = ( Base ` G ) $.
+    cycsubm.t $e |- .x. = ( .g ` G ) $.
+    cycsubm.f $e |- F = ( x e. NN0 |-> ( x .x. A ) ) $.
+    cycsubm.c $e |- C = ran F $.
+    $( Characterization of an element of the set of nonnegative integer powers
+       of an element ` A ` of a monoid.  (Contributed by AV, 28-Dec-2023.) $)
+    cycsubmel $p |- ( X e. C <-> E. i e. NN0 X = ( i .x. A ) ) $=
+      ( wcel cv wceq cn0 wrex co ovex crn cfv eleq2i wfn wb fvelrnb ax-mp oveq1
+      fnmpti fvmpt eqeq1d eqcom syl6bb rexbiia 3bitri ) IDNIGUAZNZFOZGUBZIPZFQR
+      ZIURBESZPZFQRDUPIMUCGQUDUQVAUEAQAOZBESZGVDBETLUIFQIGUFUGUTVCFQURQNZUTVBIP
+      VCVFUSVBIAURVEVBQGVDURBEUHLURBETUJUKVBIULUMUNUO $.
+
+    $d A i $.  $d B i $.  $d .x. i $.
+    $( The set of nonnegative integer powers of an element ` A ` of a monoid
+       contains ` A ` .  (Contributed by AV, 28-Dec-2023.) $)
+    cycsubmcl $p |- ( A e. B -> A e. C ) $=
+      ( vi wcel cv co wceq cn0 wrex c1 1nn0 wb oveq1 eqeq2d adantl mulg1 eqcomd
+      a1i rspcedvd cycsubmel sylibr ) BCMZBLNZBEOZPZLQRBDMUKUNBSBEOZPZLSQSQMUKT
+      UGULSPZUNUPUAUKUQUMUOBULSBEUBUCUDUKUOBCEGBHIUEUFUHABCDELFGBHIJKUIUJ $.
+
+    $d A a b j k $.  $d B a b j k x $.  $d C a b $.  $d F j k $.
+    $d G a b i j k x $.  $d .x. j k $.
+    $( The set of nonnegative integer powers of an element ` A ` of a monoid
+       forms a submonoid containing ` A ` ( see ~ cycsubmcl ), called the
+       cyclic monoid generated by the element ` A ` .  This corresponds to the
```

Comment:
Deduce this from the semigroup analogue? (later)

Reply:
If we had definitions for submagmas (only within my mathbox) and subsemigroups, we could have corresponding theorems for magmas and semigroups which could be used for the proof. Without these definitions, such theorems would be too clumsy.